tests/integration/kubernetes/confidential_common.sh

```diff
@@ -33,3 +37,29 @@ function get_remote_command_per_hypervisor() {
 
 	echo "${REMOTE_COMMAND_PER_HYPERVISOR[${KATA_HYPERVISOR}]}"
 }
+
+# This function verifies whether the input hypervisor supports confidential tests and 
+# relies on `KATA_HYPERVISOR` being an environment variable
+function check_hypervisor_for_confidential_tests(){
+	local kata_hypervisor="${1}"
+	# This check must be done with "<SPACE>${KATA_HYPERVISOR}<SPACE>" to avoid
+	# having substrings, like qemu, being matched with qemu-$something.
+	if [[ " ${SUPPORTED_TEE_HYPERVISORS[*]} " =~ " ${kata_hypervisor} " || " ${SUPPORTED_NON_TEE_HYPERVISORS[*]} " =~ " ${kata_hypervisor} " ]]; then
+        return 0
+    else
+		if [[ " ${SUPPORTED_NON_TEE_HYPERVISORS[*]} " =~ " ${kata_hypervisor} " ]]; then
+			return 2
+		fi
+        return 1
+    fi
+}
+
+# Common setup for confidential tests.
+function confidential_setup(){
+	ensure_yq
+	if ! check_hypervisor_for_confidential_tests "${KATA_HYPERVISOR}"; then
+        return 1
+    elif [[ " ${SUPPORTED_NON_TEE_HYPERVISORS[*]} " =~ " ${KATA_HYPERVISOR} " ]]; then
```

Comment:
Weird indentation cauaed by mixing tab and space indentation.

Reply:
Fixed.